x-pack/plugins/infra/server/lib/alerting/metric_threshold/preview_metric_threshold_alert.ts

```diff
@@ -159,7 +158,7 @@ export const previewMetricThresholdAlert: (
             return [a[0] + b[0], a[1] + b[1], a[2] + b[2]];
           })
       );
-      return zippedResult as any;
+      return zippedResult as number[][];
```

Comment:
What do you think about teaching the compiler what the `filter` above actually does by using

```typescript
          .filter(
            <Value>(value: Value): value is NonNullable<Value> => typeof value !== 'undefined'
          )
```

? Then it could infer the type of `zippedResults` correctly and this cast won't be necessary.